internal/parse/parse.go

```diff
@@ -323,6 +330,20 @@ func GetCacheMount(args []string, store storage.Store, imageMountLabel string, a
 		}
 	}
 
+	switch sharing {
+	case "locked":
+		// lock parent cache
+		lockfile, err := lockfile.GetLockfile(filepath.Join(newMount.Source, BuildahCacheLockfile))
+		if err != nil {
+			return newMount, errors.Wrapf(err, "Unable to acquire lock when sharing mode is locked")
+		}
+		// Will be unlocked after the RUN step is executed.
+		lockfile.Lock()
+	default:
+		// do nothing since default is `shared`
```

Comment:
Needs to reject unrecognized values for the option.  Documentation also lists `private` as an available value here.

Reply:
Yes we can error out here when value is not known. `private` is not yet implemented so it'll error out as well. I'll add `private` case here in another PR.